test/integration/deployment/util.go

```diff
@@ -221,12 +222,33 @@ func (d *deploymentTester) markAllPodsReady() {
 			d.t.Logf("failed to list Deployment pods, will retry later: %v", err)
 			return false, nil
 		}
-		if len(pods.Items) != int(*d.deployment.Spec.Replicas) {
-			d.t.Logf("%d/%d of deployment pods are created", len(pods.Items), *d.deployment.Spec.Replicas)
+		rss, err := d.c.ExtensionsV1beta1().ReplicaSets(ns).List(metav1.ListOptions{LabelSelector: selector.String()})
+		if err != nil {
+			d.t.Logf("failed to list Deployment replicasets, will retry later: %v", err)
+			return false, nil
+		}
+
+		var ownedPods []v1.Pod
+		m := make(map[types.UID]bool)
```

Comment:
@janetkuo Not sure whether there is an existing helper function that can replace map usage here.